src/utils/libertyProject.ts

```diff
@@ -79,8 +83,9 @@ export class ProjectProvider implements vscode.TreeDataProvider<LibertyProject>
 	}
 
 	// Given a list of build.gradle files, find ones that are valid to use with liberty dev-mode
-	private async findValidGradleBuildFiles(gradlePaths: string[]): Promise<string[]> {
-		const validGradleBuildFiles: string[] = [];
+	private async findValidGradleBuildFiles(gradlePaths: string[]): Promise<[string, string][]> {
+		// [gradle path, liberty project type]
+		const validGradleBuildFiles: [string, string][] = [];
```

Comment:
General comment (changes not necessary), but these arrays of arrays are going to get confusing if you need to maintain the code later on.  Consider using something like `interface` if complex data structures are needed.